app/controllers/application_controller.rb

```diff
@@ -30,9 +32,13 @@ def current_org
 
   def user_not_authorized
     if user_signed_in?
-      redirect_to plans_url, alert: _("You are not authorized to perform this action.")
+      # redirect_to plans_url, alert: _("You are not authorized to perform this action.")
+      msg = _("You are not authorized to perform this action.")
+      render_respond_to_format_with_error_message(msg, plans_url, 404, nil)
     else
-      redirect_to root_url, alert: _("You need to sign in or sign up before continuing.")
+      # redirect_to root_url, alert: _("You need to sign in or sign up before continuing.")
+      msg = "You need to sign in or sign up before continuing."
+      render_respond_to_format_with_error_message(msg, root_url, 404, nil)
```

Comment:
Should this be a 401 - unauthorized?

Reply:
Agreed.
404 = resource not found which isn't the case here.
if repeating the request after authenticating would work then should be a 401, I think.
if even after authenticating you still wouldn't be allowed the it's 403
That's my understanding anyway.